libs/algorithms/include/hpx/parallel/container_algorithms/transform.hpp

```diff
@@ -340,3 +341,112 @@ namespace hpx { namespace parallel { inline namespace v1 {
             std::forward<Proj2>(proj2));
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx { namespace ranges {
+
+    template <class I, class O>
+    using unary_transform_result = parallel::util::in_out_result<I, O>;
+
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::ranges::transform
+    HPX_INLINE_CONSTEXPR_VARIABLE struct transform_t final
+      : hpx::functional::tag<transform_t>
+    {
+    private:
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter1, typename Sent1,
+            typename FwdIter2, typename F, typename Proj = hpx::parallel::util::projection_identity,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::parallel::execution::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_iterator<FwdIter1>::value &&
+                hpx::traits::is_sentinel_for<Sent1, FwdIter1>::value &&
+                hpx::traits::is_iterator<FwdIter2>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            ranges::unary_transform_result<FwdIter1, FwdIter2>>::type
+        tag_invoke(hpx::ranges::transform_t, ExPolicy&& policy, FwdIter1 first,
+            Sent1 last, FwdIter2 dest, F&& f, Proj&& proj = Proj())
+        {
+            typedef hpx::traits::is_segmented_iterator<FwdIter1> is_segmented;
+
+            return parallel::v1::detail::transform_(
+                std::forward<ExPolicy>(policy), first, last, dest,
+                std::forward<F>(f), std::forward<Proj>(proj), is_segmented());
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename Rng, typename FwdIter,
+        typename F, typename Proj = hpx::parallel::util::projection_identity,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::parallel::execution::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_range<Rng>::value &&
+                hpx::traits::is_iterator<FwdIter>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            ranges::unary_transform_result<
+                typename hpx::traits::range_iterator<Rng>::type, FwdIter>>::type
+        tag_invoke(hpx::ranges::transform_t, ExPolicy&& policy, Rng&& rng,
+            FwdIter dest, F&& f, Proj&& proj = Proj())
+        {
+            typedef hpx::traits::is_segmented_iterator<FwdIter> is_segmented;
+            //using iterator_type =
+            //    typename hpx::traits::range_traits<Rng>::iterator_type;
+            //using is_segmented =
+            //     hpx::traits::is_segmented_iterator<iterator_type>;
```

Comment:
Are these comments needed, can they be removed?

Reply:
I kept them in purpose. Was going to ask you in the meeting right now whether they should be used or not.